### PR TITLE
new task for configuring email alerts by topic using option settings

### PR DESCRIPTION
**Pull request by tekhaus**
The genesis for this task was to create a more generalized and flexible version of a custom task developed for a merchant.

The solution here is to allow the person configuring the task to add whichever event topics they care about in the _task options_, and having each one paired with a multi-line text box that will support accessing properties of the main event object via liquid output tags.

An interesting aspect of this task is that it doesn't _seem to_ need any specific event preview code blocks, as the only action is to send an email and any object read permissions will be requested whenever the corresponding event topic is saved in the task options.

### CODE REVIEW

File: docs/send-email-alerts-by-event-topic/script.liquid [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        "action": {
+          "type": "email",
+          "options": {
+            "to": {{ email_recipients | join: ", " | json }},
+            "subject": {{ email_subject | json }},
+            "body": {{ email_body | unindent | newline_to_br | json }},
+            "from_display_name": {{ shop.name | json }}
+          }
+        }
+      }
```
Comment (isaacbowen) [inline docs/send-email-alerts-by-event-topic/script.liquid:73]: Use an action tag for this one, indenting at the same level as the lines before.

File: tasks/send-email-alerts-by-event-topic.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "Send email alerts by event topic",
+  "options": {
+    "emails_to_notify__array_required": [
+      "brad@tekha.us"
```
Comment (isaacbowen) [inline tasks/send-email-alerts-by-event-topic.json:5]: Leave this out of the task defaults; just use `null` here instead, being really explicit about the fact that you're not providing a default value, but that you are defining the presence of the option.

File: tasks/send-email-alerts-by-event-topic.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    ],
+    "general_email_subject__required": "ALERT",
+    "append_event_topic_to_email_subject__boolean": true,
+    "event_topics_and_email_bodies__keyval_multiline_required": {
+      "shopify/products/create": "Hi,\n\nProduct {{ product.id }} was created\n\nView the details in Shopify:\nOBJECT_LINK",
+      "shopify/customers/create": "Hi\n\nCustomer {{ customer.id }} was created\n\nView the details in Shopify:\nOBJECT_LINK"
```
Comment (isaacbowen) [inline tasks/send-email-alerts-by-event-topic.json:11]: 👏 Excellent defaults. :D

File: docs/send-email-alerts-by-event-topic/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+[Learn about task options in Mechanic](https://docs.usemechanic.com/article/471-task-options)
+
+## Subscriptions
+
+```liquid
+{% assign event_topics = options.event_topics_and_email_bodies__keyval_multiline_required | keys %}
+{% for event_topic in event_topics %}
+{{ event_topic }}
```
Comment (isaacbowen) [inline docs/send-email-alerts-by-event-topic/README.md:32]: Indent this, por favor? Two spaces in, per usual.

Alternatively, you could replace this forloop with the following:

```liquid
{{ event_topics | join: newline }}
```

Or, naturally, you could do this whole thing as a one-liner, using `keys` and `join` on the original option value.

Your pick. :D

**Review by isaacbowen — CHANGES_REQUESTED** [review]
_EXCELLENT_. This will be the first canonical example of generating subscriptions _entirely_ by user input. I love it. :D Thank you!

File: tasks/send-email-alerts-by-event-topic.json (hunk 2 above)
Comment (tekhaus) [reply tasks/send-email-alerts-by-event-topic.json:5]: Doh! I'm so used to running a diff against prior task code that I forgot to clean this up.

File: docs/send-email-alerts-by-event-topic/README.md (hunk 4 above)
Comment (tekhaus) [reply docs/send-email-alerts-by-event-topic/README.md:32]: For some reason I had it in my head that the subscriptions box wasn't keen on white space, but I didn't bother to check prior art here to verify.

**Review by isaacbowen — APPROVED** [review]

**Pull request merged**